src/parser/FStar.Parser.ToDocument.fs

```diff
@@ -1308,10 +1312,14 @@ and p_noSeqTerm' ps pb e = match e.tm with
       (match ret_opt with
        | None ->
         group (surround 2 1 (str "match") (p_noSeqTermAndComment false false e) (str "with"))
-       | Some ret ->
-        group (surround 2 1 (str "match")
-                            ((p_noSeqTermAndComment false false e) ^/+^ (str "returns" ^/+^ p_tmIff ret))
-                            (str "with")))
+       | Some (as_opt, ret) ->
+         group (surround 2 1 (str "match")
+                             ((p_noSeqTermAndComment false false e) ^/+^
+                              (match as_opt with
+                               | None -> empty
+                               | Some as_ident -> str " as" ^/+^ (p_ident as_ident)) ^/+^
```

Comment:
do we need the preceding " " in ` as`?

Reply:
Thanks for noticing! I removed these extra spaces and updated the pretty printing tests.